src/components/MultiSelect/MultiSelect.test.tsx

```diff
@@ -305,150 +253,117 @@ describe.skip("MultiSelect Dialog", () => {
-
-    // TESTING FOR SPACEBAR KEY
-    // Open multiselect using SPACEBAR key.
+    // // TESTING FOR SPACEBAR KEY
+    // // Open multiselect using SPACEBAR key.
     userEvent.keyboard("[Space]");
-
-    expect(screen.getByRole("dialog").getAttribute("aria-modal")).toEqual(
+    expect(screen.getByRole("button").getAttribute("aria-expanded")).toEqual(
       "true"
     );
-    expect(screen.getAllByRole("checkbox")).toHaveLength(8);
 
     // Close multiselect using SPACEBAR key.
     userEvent.keyboard("[Space]");
-
-    expect(screen.getByRole("dialog").getAttribute("aria-modal")).toBeNull();
-    expect(screen.queryByRole("checkbox")).toBeNull();
+    expect(screen.getByRole("button").getAttribute("aria-expanded")).toEqual(
+      "false"
+    );
   });
 
-  it("should allow user to close the dialog using the ESC key, then the focus should be on the MultiSelectMenuButton", () => {
+  it("should call onChange when an item without child items or a child item is selected/unselected", () => {
+    const onChangeMock = jest.fn();
+    const onMixedStateChangeMock = jest.fn();
+
     render(
       <MultiSelect
-        id="multiselect-dialog-test-id"
-        labelText="MultiSelect Label"
-        type="dialog"
+        id="multiselect-test-id"
+        helperText="multiselect-helper-text"
+        buttonText="multiselect-button-text"
+        defaultItemsVisible={defaultItemsVisible}
+        isDefaultOpen={false}
+        isSearchable={false}
+        isBlockElement={false}
         items={items}
         selectedItems={selectedTestItems}
-        onChange={() => null}
+        onChange={onChangeMock}
+        onMixedStateChange={onMixedStateChangeMock}
         onClear={() => null}
-        onApply={() => null}
       />
     );
-
-    // Initially closed
-    expect(screen.getByRole("dialog").getAttribute("aria-modal")).toBeNull();
-    expect(screen.queryByRole("checkbox")).toBeNull();
-
-    // Open multiselect.
-    userEvent.click(
-      screen.getByRole("button", {
-        name: /multiselect label/i,
-      })
-    );
-
-    expect(screen.getByRole("dialog").getAttribute("aria-modal")).toEqual(
+    // Open multiselect menu.
+    userEvent.click(screen.queryByRole("button"));
+    expect(screen.queryByRole("button").getAttribute("aria-expanded")).toEqual(
       "true"
     );
-    expect(screen.getAllByRole("checkbox")).toHaveLength(8);
-
-    // Close the diaog using the ESC key.
-    userEvent.keyboard("{Escape}");
-
-    expect(screen.getByRole("dialog").getAttribute("aria-modal")).toBeNull();
-    expect(screen.queryByRole("checkbox")).toBeNull();
-    expect(
-      screen.getByRole("button", {
-        name: /multiselect label/i,
-      })
-    ).toHaveFocus();
-  });
-
-  it("should call onChange when an item without child items or a child item is selected/unselected", () => {
-    const onChangeMock = jest.fn();
-    const onMixedStateChangeMock = jest.fn();
 
     render(
```

Comment:
In this case, try to use `rerender` because using `render` twice puts two `MultiSelect` components in the testing DOM while `rerender` will update the existing instance.